crates/apollo-compiler/tests/input_coercion.rs

```diff
@@ -0,0 +1,149 @@
+use apollo_compiler::request::coerce_variable_values;
+use apollo_compiler::ExecutableDocument;
+use apollo_compiler::Schema;
+use serde_json_bytes::ByteString;
+use serde_json_bytes::Map;
+use serde_json_bytes::Value;
+
+///
+/// Builds and coerces a GraphQL mutation variable map for testing Float coercion behavior.
+///
+/// Helper function, used in unit tests to verify how GraphQL variable coercion behaves when provided
+/// with different numeric types or extreme values.
+///
+/// It defines a minimal GraphQL schema with a `Car` type and a corresponding `CarInput` that
+/// contains two `Float!` fields: `range` and `totalKilometers`.
+/// The function then constructs a mutation using these input types and attempts to coerce the provided
+/// values into the correct GraphQL variable format.
+///
+/// # Type Parameters
+/// * `R` – The type of the `range` argument, must implement [`Into<f64>`].
+/// * `K` – The type of the `total_kilometers` argument, must implement [`Into<f64>`].
+///
+/// # Arguments
+/// * `range` – A numeric value (or convertible type) representing the range attribute of a car.
+/// * `total_kilometers` – A numeric value (or convertible type) representing the total kilometers of a car.
+///
+/// # Returns
+/// A [`Result`] containing the coerced [`Map<ByteString, Value>`] representing the
+/// GraphQL variable map if coercion succeeds.
+/// If coercion fails (for example, due to exceeding `f64` limits or type mismatches),
+/// an error message is returned as a [`String`].
+fn build_and_coerce_test_mutation_variables<R, K>(
+    range: R,
+    total_kilometers: K,
+) -> Result<Map<ByteString, Value>, String>
+where
+    R: Into<f64>,
+    K: Into<f64>,
+{
```

Comment:
This helper function is too complex. What you had previously with a single test was more readable and covered the changes more appropriately. If you go back to what you had previously, you can just add additional calls to `coerce_variable_values` with different values and additional asserts to validate .